api-lib/handleUserAddedMsg.ts

```diff
@@ -0,0 +1,80 @@
+import { isFeatureEnabled } from '../src/config/features';
+
+import * as queries from './gql/queries';
+import {
+  Channels,
+  DiscordUserAddedOrRemoved,
+  sendSocialMessage,
+} from './sendSocialMessage';
+import { Awaited } from './ts4.5shim';
+import { EventTriggerPayload } from './types';
+
+type GetChannelsProps = {
+  data: EventTriggerPayload<'users', 'INSERT' | 'UPDATE'>['event']['data'];
+  circle: Awaited<ReturnType<typeof queries.getCircle>>['circles_by_pk'];
+  profiles: Awaited<
+    ReturnType<typeof queries.getProfileAndMembership>
+  >['profiles'];
+  channels: Channels<DiscordUserAddedOrRemoved>;
+};
+
+function getChannels(
+  props: GetChannelsProps
+): Channels<DiscordUserAddedOrRemoved> {
+  const { channels, circle, data, profiles } = props || {};
+
+  const { discord_channel_id: channelId, discord_role_id: roleId } =
+    circle?.discord_circle || {};
+
+  if (
+    channels?.isDiscordBot &&
+    isFeatureEnabled('discord') &&
+    channelId &&
+    roleId
+  ) {
+    const user = profiles[0].user;
+
+    return {
+      isDiscordBot: true,
+      discordBot: {
+        type: 'user-added',
+        channelId,
+        roleId,
+        discordId: user?.user_snowflake,
+        address: data.new.address,
+        circleName: circle?.name ?? 'Unknown',
```

Comment:
followup: talked this over with @reeserj and we're removing this feature from the current set of deliverables, since it's not supported in our core app at this time.